xspec/xspec
# Basic test for code coverage (command line)

This pull request derives from #300. So needs to be handled after that.

---

This pull request adds some basic tests for the code coverage on the command line.

A [`TODO`](https://github.com/AirQuick/xspec/blob/8b09290f73305e70b10be3a8c778fbad9db6c93d/test/xspec.bats#L158-L159) line is just for fixing #208 in the future. (Replacing `up and down` with `up & down` will reproduce #208.)

### How to test

Here's what I tested on my local machine.

1. Do `set SAXON_JAR=\path\to\saxon9ee.jar`. (`saxon9pe.jar` should be fine too, but I haven't tested.)
1. Run `run-bats.cmd` in `test/`.
    Must pass, including the test case 14.

1. Do `set SAXON_JAR=\path\to\saxon9he.jar`.
1. Run `run-bats.cmd`.
    Must pass, skipping the test case 14. (demonstrated on [Travis](https://travis-ci.org/AirQuick/xspec/jobs/419519679#L1196) and [AppVeyor](https://ci.appveyor.com/project/AirQuick/xspec/build/776-test_cmdline-coverage/job/lix14w35il4c0ljm#L144))

1. Check out 6e0702a78d6a540eb843f4dae4132ebe2ec91541 which is intentionally broken.
1. Do `set SAXON_JAR=\path\to\saxon9ee.jar`.
1. Run `run-bats.cmd`.
    At least 14 and 15 must fail. (15 is demonstrated on [Travis](https://travis-ci.org/AirQuick/xspec/jobs/419528635#L1197-L1223) and [AppVeyor](https://ci.appveyor.com/project/AirQuick/xspec/build/780-test_cmdline-coverage/job/dri7tqts29bqr9wg#L146))

Repeat the same steps using `run-bats.sh --tap` on Ubuntu.
